<?xml version="1.0"?>
<launch>
  <arg name="joy_dev" default="/dev/input/js0" />

  <node pkg="crazyflie_driver" type="crazyflie_server" name="crazyflie_server" output="screen">
    <rosparam>
      broadcast_uri: "radio://0/80/2M/FFE7E7E7E7"
      world_frame: "/world"
      num_cfs: 2
      crazyflie1:
        uri: "radio://0/80/2M/E7E7E7E701"
        frame: "/crazyflie1"
        id: 1
      crazyflie2:
        uri: "radio://0/80/2M/E7E7E7E702"
        frame: "/crazyflie2"
        id: 2
    </rosparam>
  </node>

  <node name="joy" pkg="joy" type="joy_node" output="screen">
    <param name="dev" value="$(arg joy_dev)" />
  </node>

  <node pkg="crazyflie_manager" type="crazyflie_manager" name="crazyflie_manager" output="screen">
    <param name="num_cfs" value="2" />
    <param name="crazyflie1/csv_file" value="$(find crazyflie_demo)/scripts/trajectory1.csv" />
    <param name="crazyflie1/x_offset" value="0.0" />
    <param name="crazyflie1/y_offset" value="0.0" />
    <param name="crazyflie2/csv_file" value="$(find crazyflie_demo)/scripts/trajectory1.csv" />
    <param name="crazyflie2/x_offset" value="0.0" />
    <param name="crazyflie2/y_offset" value="-0.75" />
  </node>

  <!-- run vicon bridge -->
  <include file="$(find vicon_bridge)/launch/vicon.launch"/>

  <node name="object_tracker" pkg="object_tracker" type="object_tracker_node" output="screen">
    <rosparam command="load" file="$(find crazyflie_demo)/launch/cf2.yaml" />
  </node>

</launch>
